src/transformers/models/dpt/modeling_dpt.py

```diff
@@ -0,0 +1,1130 @@
+        in_channels (int): ViT feature channels. Default: 768. out_channels (List): output channels of each stage.
+            Default: [96, 192, 384, 768].
+        readout_type (str): Type of readout operation. Default: 'ignore'. patch_size (int): The patch size. Default:
+        16. init_cfg (dict, optional): Initialization config dict. Default: None.
+    """
+
+    def __init__(self, config):
+        super().__init__()
+
+        self.config = config
+        out_channels = config.post_process_channels
+
+        self.projects = nn.ModuleList(
+            [
+                nn.Conv2d(
+                    in_channels=config.hidden_size,
+                    out_channels=out_channel,
+                    kernel_size=1,
+                )
+                for out_channel in out_channels
+            ]
+        )
+
+        self.resize_layers = nn.ModuleList(
+            [
+                nn.ConvTranspose2d(
+                    in_channels=out_channels[0], out_channels=out_channels[0], kernel_size=4, stride=4, padding=0
+                ),
+                nn.ConvTranspose2d(
+                    in_channels=out_channels[1], out_channels=out_channels[1], kernel_size=2, stride=2, padding=0
+                ),
+                nn.Identity(),
+                nn.Conv2d(
+                    in_channels=out_channels[3], out_channels=out_channels[3], kernel_size=3, stride=2, padding=1
+                ),
+            ]
+        )
+        if config.readout_type == "project":
+            self.readout_projects = nn.ModuleList()
+            for _ in range(len(self.projects)):
+                self.readout_projects.append(
+                    nn.Sequential(nn.Linear(2 * config.hidden_size, config.hidden_size), ACT2FN[config.hidden_act])
+                )
+
+    def forward(self, inputs):
+        """
+        Inputs: list of torch.FloatTensor, each of shape (B, L + 1, C).
+        """
+        assert isinstance(inputs, list)
+        out = []
+
+        for i, x in enumerate(inputs):
+            # reshape to (B, C, H, W)
+            x, cls_token = x[:, 1:], x[:, 0]
+            B, L, C = x.shape
+            size = int(math.sqrt(L))
+            x = x.reshape(B, size, size, C)
+            x = x.permute(0, 3, 1, 2).contiguous()
+
+            feature_shape = x.shape
+            if self.config.readout_type == "project":
+                x = x.flatten(2).permute((0, 2, 1))
+                readout = cls_token.unsqueeze(1).expand_as(x)
+                x = self.readout_projects[i](torch.cat((x, readout), -1))
+                x = x.permute(0, 2, 1).reshape(feature_shape)
+            elif self.config.readout_type == "add":
+                x = x.flatten(2) + cls_token.unsqueeze(-1)
+                x = x.reshape(feature_shape)
+            else:
+                pass
+            x = self.projects[i](x)
+            x = self.resize_layers[i](x)
+            out.append(x)
+
+        return out
+
+
+class DPTPreActResidualConvUnit(nn.Module):
```

Comment:
This names could be better chosen.

Reply:
`DPTPreActResidualConvUnit` -> `DPTPreActResidualLayer`